src/Aspire.Hosting.Orleans/OrleansServiceClientExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Orleans;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extension methods for <see cref="OrleansServiceClient"/>.
+/// </summary>
+public static class OrleansServiceClientExtensions
+{
+    /// <summary>
+    /// Adds an Orleans client to the resource.
+    /// </summary>
+    /// <param name="builder">The builder on which add the Orleans service builder.</param>
+    /// <param name="orleansServiceClient">The Orleans service client, containing clustering, etc.</param>
+    /// <returns>The resource builder.</returns>
+    /// <exception cref="InvalidOperationException">Clustering has not been configured.</exception>
+    public static IResourceBuilder<T> WithReference<T>(
```

Comment:
When would I use `WithReference(OrleansServiceClient)` over using `WithReference(OrleansService)`?

Is it confusing to have both? What's the difference between:

```C#
builder.AddProject<Projects.OrleansServer>("silo")
       .WithReference(orleans);
```

and 
```C#
builder.AddProject<Projects.OrleansClient>("frontend")
       .WithReference(orleans.AsClient());
```

?

Reply:
You use the service one for projects with an Orleans server and the client one with the Orleans clients. Their configuration follows the same format, but servers have more (storage, etc)